src/components/table-hoc/TableWithPredicate.tsx

```diff
@@ -0,0 +1,78 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {ReduxConnect} from '../../utils/ReduxUtils';
+import {IItemBoxProps} from '../itemBox/ItemBox';
+import {SingleSelectConnected} from '../select/SingleSelectConnected';
+import * as styles from './styles/TableWithPredicates.scss';
+import {ITableHOCOwnProps} from './TableHOC';
+import {TableHOCUtils} from './TableHOCUtils';
+
+export interface ITableWithPredicateConfig {
+    id: string;
+    values: IItemBoxProps[];
+    prepend?: React.ReactNode;
+    matchPredicate?: (predicate: string, datum: any) => boolean;
+    isServer?: boolean;
+}
+
+export interface ITableWithPredicateStateProps {
+    predicate: string;
+}
+
+export interface ITableWithPredicateProps extends Partial<ITableWithPredicateStateProps>,
+    ITableHOCOwnProps {}
+
+const TableWithPredicatePropsToOmit = keys<ITableWithPredicateStateProps>();
+
+const defaultMatchPredicate = (predicate: string, datum: any) => !predicate || _.some(_.values(datum), (value: string) => value === predicate);
+
+type TableWithPredicateComponent = React.ComponentClass<ITableWithPredicateProps>;
+
+export const tableWithPredicate = (config: ITableWithPredicateConfig) => (Component: TableWithPredicateComponent): TableWithPredicateComponent => {
+
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableWithPredicateProps): ITableWithPredicateStateProps | ITableHOCOwnProps => {
+        const listbox = _.findWhere(state.listBoxes, {id: TableHOCUtils.getPredicateId(ownProps.id, config.id)});

```

Comment:
little change but you can reuse the selector "getListBox", 
`getListBox(state, {id: TableHOCUtils.getPredicateId(ownProps.id, config.id)})`